src/audio/kpb.c

```diff
@@ -1315,6 +1318,62 @@ static enum task_state kpb_draining_task(void *arg)
 	return SOF_TASK_STATE_COMPLETED;
 }
 
+/**
+ * \brief Copy s16 data linear to circular
+ *
+ * \param[in] source - source buffer
+ * \param[in] ostream - output stream
+ * \param[in] bytes_total - number of bytes to copy
+ *
+ * \return none.
+ */
+static inline void drain_s16(int16_t *source, struct audio_stream *ostream, size_t bytes_total)
+{
+	int16_t *src = source;
+	int16_t *dst = ostream->w_ptr;
+	int bmax;
+	int b;
+	int bytes_copied = 0;
+
+	while (bytes_copied < bytes_total) {
+		b = bytes_total - bytes_copied;
+		bmax = audio_stream_bytes_without_wrap(ostream, dst);
+		b = MIN(b, bmax);
+		memcpy(dst, src, b);
+		src += b >> BYTES_TO_S16_SAMPLES;
+		dst = audio_stream_wrap(ostream, dst + (b >> BYTES_TO_S16_SAMPLES));
```

Comment:
similar to another PR: `src` and `dst` could be `uint8_t` to avoid shifting

Reply:
I used in audio_stream.h the char * type since one existing similar function used it. Hope it's OK.

Reply:
Now uint8_t